src/components/data/DataTable.tsx

```diff
@@ -81,23 +86,25 @@ export default function DataTable<Row>({
   });
 
   const noContent = loading || (!rows.length && emptyMessage);
-  const fields = columns.map(column => column.field);
+  const fields = useMemo(() => columns.map(column => column.field), [columns]);
 
-  function selectRow(row: Row) {
+  const selectRow = useStableCallback((row: Row) => {
     onSelectRow?.(row);
-  }
-
-  function confirmRow(row: Row) {
+  });
+  const confirmRow = useStableCallback((row: Row) => {
     onConfirmRow?.(row);
-  }
```

Comment:
Wrap callback props so we don't unnecessarily re-render even if consumers fail to `useCallback` these.